lib/chef/audit/runner.rb

```diff
@@ -0,0 +1,250 @@
+autoload :Inspec, "inspec"
+
+require_relative "default_attributes"
+require_relative "reporter/audit_enforcer"
+require_relative "reporter/automate"
+require_relative "reporter/chef_server_automate"
+require_relative "reporter/json_file"
+
+class Chef
+  module Audit
+    class Runner < EventDispatch::Base
+      extend Forwardable
+
+      attr_accessor :run_id, :recipes
+      attr_reader :node
+      def_delegators :node, :logger
+
+      def enabled?
+        audit_cookbook_present = recipes.include?("audit::default")
+
+        logger.info("#{self.class}##{__method__}: inspec profiles? #{inspec_profiles.any?}")
+        logger.info("#{self.class}##{__method__}: audit cookbook? #{audit_cookbook_present}")
+
+        inspec_profiles.any? && !audit_cookbook_present
+      end
+
+      def node=(node)
+        @node = node
+        node.default["audit"] = Chef::Audit::DEFAULT_ATTRIBUTES.merge(node["audit"] || {})
```

Comment:
I think you should be able to `node.default["audit"] = Chef::Audit::DEFAULT_ATTRIBUTES`.

If you want to go this way you should also probably not merge with the merged `node["audit"]` but `merge(node.default["audit"])` instead, but at the same time I'm fairly sure it'll be okay to assume that is always blank here.  And because roles/envs/policyfiles wind up in different mashes you don't have to worry about those concerns at all (although I think that happens later anyway).

Reply:
I experimented with this a bit following our conversation in standup.

Doing something like this does not work:

```
def node_load_success(node)
  node.default["audit"] = Chef::Audit::DEFAULT_ATTRIBUTES
end
```

It appears the attributes have not yet been loaded, so when that does happen the defaults are overwritten and I see a nil exception because `node.default["json_file"]["location"]` is unset (ie, the defaults are not present).

Something like this does work:

```
def node_load_completed(node)
  node.default["audit"] = Chef::Audit::DEFAULT_ATTRIBUTES.merge(node.default["audit"] || {})
end
```

The attributes have already been applied, so merging those into the defaults results in the defaults being set with the user's settings applied over the top.

Unless there's something I'm missing this seems like the way this needs to happen.

Reply:
I'll see if I can remove the `|| {}`, I think that was necessary because of me doing something wrong early in my testing.